Allow the use of toga.Window as the main window (#2649)
beeware/toga: docs/reference/api/mainwindow.rst

-----

A :class:`toga.MainWindow` is a :class:`toga.Window` that can serve as the main
interface to an application. A :class:`toga.MainWindow` may optionally have a toolbar.
The presentation of :class:`toga.MainWindow` is platform dependent:

> * On desktop platforms that place menus inside windows (e.g., Windows, and most Linux
>   window managers), a :class:`toga.MainWindow` instance will display a menu bar that
>   contains the app control commands (such as About, Quit, and anything else required by
>   the platform's HIG).

Reviewer: This makes user-defined commands seem like an afterthought. In fact, I don't think it's helpful to distinguish between HIG commands and user-defined commands here – that can be done on the App or Commands pages.

```suggestion
* On desktop platforms that place menus inside windows (e.g., Windows, and most Linux
  window managers), a :class:`toga.MainWindow` instance will display a menu bar that
  contains :attr:`App.commands`.
```

* On desktop platforms that use an app-level menu bar (e.g., macOS, and some Linux
  window managers), the window will not have a menu bar; all menu items will be
  displayed in the app bar.

* On mobile, web and console platforms, a :class:`toga.MainWindow` will include a title
  bar that can contain both menus and toolbar items.

In addition to the platform's default commands, user-defined commands can be

> added to the :class:`toga.MainWindow`'s menu by adding them to

Reviewer: I can't post a suggestion on an unmodified line, but if the previous suggestion is accepted, this sentence should be removed.
